pyramda/iterable/flatten.py

```diff
@@ -1,37 +1,30 @@
 from collections import Iterable
 from past.builtins import basestring
 
+from pyramda.isinstance import isinstance
 from pyramda.function.curry import curry
 
 
 @curry
-def flatten(xs, depth):
+def flatten_until(is_leaf, xs):
```

Comment:
I agree.  With the current implementation I have, `flatten_until(isinstance(basestring), 5)`  will return  `5` instead of `[5]`. It is definitely better to return  `[5]` because in this way there is no  ambiguity between whether flatten_until returns a list or a non-list (it will always return a list irrespective of whether the inputs are iterable or not).

One a  related note, how should flatten behave if `xs` happens to be a nested dictionary? Should it just raise an exception if it encounters a Mapping type?  The currently implementation I have just returns the top level keys of the dictionary which is not really correct I think:
```python
flatten({'a':1 , 'b': {'a':2}, 'c':{'d':4}})
# ['a', 'b', 'c']
```